pkg/extension/delete_interceptor.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extension
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Peripli/service-manager/pkg/types"
+
+	"github.com/Peripli/service-manager/pkg/query"
+
+	"github.com/Peripli/service-manager/storage"
+)
+
+type namedDeleteAPIFunc struct {
+	Name string
+	Func func(InterceptDeleteOnAPI) InterceptDeleteOnAPI
+}
+
+type namedDeleteTxFunc struct {
+	Name string
+	Func func(InterceptDeleteOnTx) InterceptDeleteOnTx
+}
+
+type deleteHookOnAPIHandler struct {
+	DeleteHookOnAPIFuncs []*namedDeleteAPIFunc
```

Comment:
do we still have Hook as concept ? but the field name and the object name are confusing